sway/main.c

```diff
@@ -186,16 +186,12 @@ static void log_kernel(void) {
 
 static bool drop_permissions(void) {
 	if (getuid() != geteuid() || getgid() != getegid()) {
-		if (setgid(getgid()) != 0) {
-			sway_log(SWAY_ERROR, "Unable to drop root, refusing to start");
-			return false;
-		}
-		if (setuid(getuid()) != 0) {
+		if (setuid(getuid()) != 0 || setgid(getgid()) != 0) {
 			sway_log(SWAY_ERROR, "Unable to drop root, refusing to start");
 			return false;
 		}
 	}
-	if (setuid(0) != -1) {
+	if (setuid(0) == 0) {
```

Comment:
-1. It's better to enter the error condition if something very bad happens, like `setuid` returning 42. Additionally, the spec says that if `setuid` fails, it must return -1, but doesn't say that if `setuid` succeeds, it should return 0.

Reply:
It does say that if it succeeds, it should return 0, at least in the man page.

Reply:
Yeah, sorry, skipped the important bit.